DNN test failures with CuDNN support

### System Information

OpenCV version: 4.6.0
Operating System / Platform: Ubuntu 20.04
Compiler & compiler version: GCC 9.4.0

### Detailed description

There are several failed DNN tests if to build OpenCV with CuDNN (8.6.0 version) support.

The pipeline log is [here](https://github.com/opencv/ci-gha-workflow/actions/runs/3226292036/jobs/5279631322#step:16:18146).

Failed tests:
```
[  FAILED  ] 40 tests, listed below:
[  FAILED  ] Test_Model.TextRecognition/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_Model.TextRecognition/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Convolution3D/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Convolution3D_bias/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Shape/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Shape/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Power/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Power/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Elementwise_Log/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Elementwise_Log/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Elementwise_not/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Elementwise_not/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Compare_EQ/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Compare_EQ/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Compare_GT/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Compare_GT/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Compare_LT/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Compare_LT/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Compare_GTorEQ/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Compare_GTorEQ/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Compare_LEorEQ/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Compare_LEorEQ/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Eltwise3D/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Eltwise3D/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.PoolConv3D/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Div/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Div/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.Split/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.Split/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_layers.LSTM_Activations/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.SubFromConst/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_layers.SubFromConst/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_nets.LResNet100E_IR/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_nets.Emotion_ferplus/0, where GetParam() = CUDA/CUDA
[  FAILED  ] Test_ONNX_nets.Emotion_ferplus/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_ONNX_nets.Resnet34_kinetics/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_TensorFlow_layers.Convolution3D/1, where GetParam() = CUDA/CUDA_FP16
[  FAILED  ] Test_TensorFlow_layers.concat_3d/1, where GetParam() = CUDA/CUDA_FP16
```

<details><summary>More details about each of them:</summary>
<p>

```
[ RUN      ] Test_Model.TextRecognition/0, where GetParam() = CUDA/CUDA
/home/ci/opencv/modules/dnn/test/test_model.cpp:148: Failure
Expected equality of these values:
  results[0]
    Which is: ""
  seq
    Which is: "welcome"
ROI[0]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:149: Failure
Expected equality of these values:
  results[1]
    Which is: ""
  seq
    Which is: "welcome"
ROI[1]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
[  FAILED  ] Test_Model.TextRecognition/0, where GetParam() = CUDA/CUDA (120 ms)

---

[ RUN      ] Test_Model.TextRecognition/1, where GetParam() = CUDA/CUDA_FP16
/home/ci/opencv/modules/dnn/test/test_model.cpp:140: Failure
Expected equality of these values:
  result
    Which is: ""
  seq
    Which is: "welcome"
Full frame: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:148: Failure
Expected equality of these values:
  results[0]
    Which is: ""
  seq
    Which is: "welcome"
ROI[0]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:149: Failure
Expected equality of these values:
  results[1]
    Which is: ""
  seq
    Which is: "welcome"
ROI[1]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
[  FAILED  ] Test_Model.TextRecognition/1, where GetParam() = CUDA/CUDA_FP16 (701 ms)

---

[ RUN      ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/0, where GetParam() = CUDA/CUDA
/home/ci/opencv/modules/dnn/test/test_model.cpp:140: Failure
Expected equality of these values:
  result
    Which is: ""
  seq
    Which is: "welcome"
Full frame: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:148: Failure
Expected equality of these values:
  results[0]
    Which is: ""
  seq
    Which is: "welcome"
ROI[0]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:149: Failure
Expected equality of these values:
  results[1]
    Which is: ""
  seq
    Which is: "welcome"
ROI[1]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
[  FAILED  ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/0, where GetParam() = CUDA/CUDA (89 ms)

---

[ RUN      ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/1, where GetParam() = CUDA/CUDA_FP16
/home/ci/opencv/modules/dnn/test/test_model.cpp:140: Failure
Expected equality of these values:
  result
    Which is: ""
  seq
    Which is: "welcome"
Full frame: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:148: Failure
Expected equality of these values:
  results[0]
    Which is: ""
  seq
    Which is: "welcome"
ROI[0]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
/home/ci/opencv/modules/dnn/test/test_model.cpp:149: Failure
Expected equality of these values:
  results[1]
    Which is: ""
  seq
    Which is: "welcome"
ROI[1]: /home/ci/opencv_extra/testdata/dnn/text_rec_test.png
[  FAILED  ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/1, where GetParam() = CUDA/CUDA_FP16 (710 ms)

---

[ RUN      ] Test_ONNX_layers.Convolution3D/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/cudnn/convolution.hpp:303: error: (-217:Gpu API call) cuDNN did not return a suitable algorithm for convolution. in function 'ConvolutionAlgorithm'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Convolution3D/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.Convolution3D_bias/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/cudnn/convolution.hpp:303: error: (-217:Gpu API call) cuDNN did not return a suitable algorithm for convolution. in function 'ConvolutionAlgorithm'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Convolution3D_bias/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.Shape/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Shape/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Shape/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Shape/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Power/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Power/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Power/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Power/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.Elementwise_Log/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Elementwise_Log/0, where GetParam() = CUDA/CUDA (3 ms)

---

[ RUN      ] Test_ONNX_layers.Elementwise_Log/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Elementwise_Log/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Elementwise_not/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Elementwise_not/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Elementwise_not/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Elementwise_not/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_EQ/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_EQ/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_EQ/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_EQ/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_GT/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_GT/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_GT/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_GT/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_LT/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_LT/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_LT/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_LT/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_GTorEQ/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_GTorEQ/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_GTorEQ/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_GTorEQ/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_LEorEQ/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_LEorEQ/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Compare_LEorEQ/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Compare_LEorEQ/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Eltwise3D/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Eltwise3D/0, where GetParam() = CUDA/CUDA (4 ms)

---

[ RUN      ] Test_ONNX_layers.Eltwise3D/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Eltwise3D/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.PoolConv3D/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/cudnn/convolution.hpp:303: error: (-217:Gpu API call) cuDNN did not return a suitable algorithm for convolution. in function 'ConvolutionAlgorithm'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.PoolConv3D/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_layers.Div/0, where GetParam() = CUDA/CUDA
FALLBACK: Layer [NaryEltwise]:[onnx_node_output_0!2] is expected to has backend implementation
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/test/test_common.hpp:214: error: (-213:The function/feature is not implemented) Implementation fallbacks are not expected in this test in function 'expectNoFallbacks'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Div/0, where GetParam() = CUDA/CUDA (3 ms)

---

[ RUN      ] Test_ONNX_layers.Div/1, where GetParam() = CUDA/CUDA_FP16
FALLBACK: Layer [NaryEltwise]:[onnx_node_output_0!2] is expected to has backend implementation
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/test/test_common.hpp:214: error: (-213:The function/feature is not implemented) Implementation fallbacks are not expected in this test in function 'expectNoFallbacks'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Div/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)

---

[ RUN      ] Test_ONNX_layers.Split/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Split/0, where GetParam() = CUDA/CUDA (16 ms)

---

[ RUN      ] Test_ONNX_layers.Split/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Split/1, where GetParam() = CUDA/CUDA_FP16 (17 ms)

---

[ RUN      ] Test_ONNX_layers.LSTM_Activations/0, where GetParam() = CUDA/CUDA
/home/ci/opencv/modules/dnn/test/test_common.impl.hpp:74: Failure
Expected: (normL1) <= (l1), actual: 0.0435941 vs 1e-05
  |ref| = 0.16841614246368408
/home/ci/opencv/modules/dnn/test/test_common.impl.hpp:77: Failure
Expected: (normInf) <= (lInf), actual: 0.168416 vs 0.0001
  |ref| = 0.16841614246368408
[  FAILED  ] Test_ONNX_layers.LSTM_Activations/0, where GetParam() = CUDA/CUDA (5 ms)

---

[ RUN      ] Test_ONNX_layers.SubFromConst/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.SubFromConst/0, where GetParam() = CUDA/CUDA (3 ms)

---

[ RUN      ] Test_ONNX_layers.SubFromConst/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.SubFromConst/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_ONNX_nets.LResNet100E_IR/1, where GetParam() = CUDA/CUDA_FP16
/home/ci/opencv/modules/dnn/test/test_common.impl.hpp:74: Failure
Expected: (normL1) <= (l1), actual: 0.00869873 vs 0.008
  |ref| = 3.3252670764923096
[  FAILED  ] Test_ONNX_nets.LResNet100E_IR/1, where GetParam() = CUDA/CUDA_FP16 (1887 ms)

---

[ RUN      ] Test_ONNX_nets.Emotion_ferplus/0, where GetParam() = CUDA/CUDA
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_nets.Emotion_ferplus/0, where GetParam() = CUDA/CUDA (76 ms)

---

[ RUN      ] Test_ONNX_nets.Emotion_ferplus/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/tensor_ops.hpp:43: error: (-215:Assertion failed) is_shape_same(dest, src) in function 'copy'
" thrown in the test body.
[  FAILED  ] Test_ONNX_nets.Emotion_ferplus/1, where GetParam() = CUDA/CUDA_FP16 (72 ms)

---

[ RUN      ] Test_ONNX_nets.Resnet34_kinetics/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/cudnn/convolution.hpp:303: error: (-217:Gpu API call) cuDNN did not return a suitable algorithm for convolution. in function 'ConvolutionAlgorithm'
" thrown in the test body.
[  FAILED  ] Test_ONNX_nets.Resnet34_kinetics/1, where GetParam() = CUDA/CUDA_FP16 (522 ms)

---

[ RUN      ] Test_TensorFlow_layers.Convolution3D/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/cudnn/convolution.hpp:303: error: (-217:Gpu API call) cuDNN did not return a suitable algorithm for convolution. in function 'ConvolutionAlgorithm'
" thrown in the test body.
[  FAILED  ] Test_TensorFlow_layers.Convolution3D/1, where GetParam() = CUDA/CUDA_FP16 (3 ms)

---

[ RUN      ] Test_TensorFlow_layers.concat_3d/1, where GetParam() = CUDA/CUDA_FP16
unknown file: Failure
C++ exception with description "OpenCV(4.6.0-dev) /home/ci/opencv/modules/dnn/src/layers/../cuda4dnn/primitives/../csl/cudnn/convolution.hpp:303: error: (-217:Gpu API call) cuDNN did not return a suitable algorithm for convolution. in function 'ConvolutionAlgorithm'
" thrown in the test body.
[  FAILED  ] Test_TensorFlow_layers.concat_3d/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)
```

</p>
</details>

### Steps to reproduce

Build OpenCV with CUDA and CuDNN:

1. Clone OpenCV
```
git clone --branch 4.x https://github.com/opencv/opencv.git
```
2. Clone OpenCV Contrib
```
git clone --branch 4.x https://github.com/opencv/opencv_contrib.git
```
3. Clone OpenCV Extra for tests
```
git clone --branch 4.x https://github.com/opencv/opencv_extra.git
```
4. Generate OpenCV with flags
```
cmake -B build -DBUILD_DOCS=ON -DPYTHON_DEFAULT_EXECUTABLE=/usr/bin/python3 -DBUILD_EXAMPLES=ON -DOPENCV_ENABLE_NONFREE=ON -DWITH_CUDA=ON -DBUILD_LIST=dnn,cudev,ts -DCUDA_ARCH_BIN=6.1 -DWITH_VA=OFF -DOPENCV_EXTRA_MODULES_PATH=opencv_contrib/modules opencv
```
5. Build OpenCV
```
cmake --build build --target install -j $(nproc)
```
6. Set `OPENCV_TEST_DATA_PATH` env variable:
```
export OPENCV_TEST_DATA_PATH=$(pwd)/opencv_extra/testdata
```
7. Run DNN tests
```
cd build && ./bin/opencv_test_dnn
```

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/dnn/src/layers/elementwise_layers.cpp
+++ b/modules/dnn/src/layers/elementwise_layers.cpp
@@ -1108,7 +1108,7 @@ struct CeilFunctor : public BaseDefaultFunctor<CeilFunctor>
 
     bool supportBackend(int backendId, int)
     {
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE;
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA || backendId == DNN_BACKEND_HALIDE;
     }
 
     inline float calculate(float x) const
@@ -1143,7 +1143,7 @@ struct FloorFunctor : public BaseDefaultFunctor<FloorFunctor>
 
     bool supportBackend(int backendId, int)
     {
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE;
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA || backendId == DNN_BACKEND_HALIDE;
     }
 
     inline float calculate(float x) const
@@ -1178,7 +1178,7 @@ struct LogFunctor : public BaseDefaultFunctor<LogFunctor>
 
     bool supportBackend(int backendId, int)
     {
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE;
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA || backendId == DNN_BACKEND_HALIDE;
     }
 
     inline float calculate(float x) const
@@ -1213,7 +1213,7 @@ struct RoundFunctor : public BaseDefaultFunctor<RoundFunctor>
 
     bool supportBackend(int backendId, int)
     {
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE;
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA || backendId == DNN_BACKEND_HALIDE;
     }
 
     inline float calculate(float x) const
@@ -1253,7 +1253,7 @@ struct SqrtFunctor : public BaseDefaultFunctor<SqrtFunctor>
 
     bool supportBackend(int backendId, int)
     {
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE;
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA || backendId == DNN_BACKEND_HALIDE;
     }
 
     inline float calculate(float x) const
@@ -1295,7 +1295,7 @@ struct NotFunctor : public BaseDefaultFunctor<NotFunctor>
 
     bool supportBackend(int backendId, int)
     {
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE;
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA || backendId == DNN_BACKEND_HALIDE;
     }
 
     inline float calculate(float x) const

--- a/modules/dnn/src/onnx/onnx_importer.cpp
+++ b/modules/dnn/src/onnx/onnx_importer.cpp
@@ -2971,6 +2971,8 @@ void ONNXImporter::parseElementWise(LayerParams& layerParams, const opencv_onnx:
                 LayerParams constParams;
                 constParams.name = node_proto.input(i);
                 constParams.type = "Const";
+                // Non-constant propagated layers cannot output 1-d or 0-d tensors.
+                inp.dims = std::max(inp.dims, 2);
                 constParams.blobs.push_back(inp);
 
                 opencv_onnx::NodeProto proto;

--- a/modules/dnn/test/test_onnx_importer.cpp
+++ b/modules/dnn/test/test_onnx_importer.cpp
@@ -221,11 +221,21 @@ TEST_P(Test_ONNX_layers, GatherMulti)
 
 TEST_P(Test_ONNX_layers, Convolution3D)
 {
+    if (backend == DNN_BACKEND_CUDA && target == DNN_TARGET_CUDA_FP16)
+    {
+        // CUDA_FP16: cuDNN did not return a suitable algorithm for convolution.
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_CUDA_FP16);
+    }
     testONNXModels("conv3d");
 }
 
 TEST_P(Test_ONNX_layers, Convolution3D_bias)
 {
+    if (backend == DNN_BACKEND_CUDA && target == DNN_TARGET_CUDA_FP16)
+    {
+        // CUDA_FP16: cuDNN did not return a suitable algorithm for convolution.
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_CUDA_FP16);
+    }
     testONNXModels("conv3d_bias");
 }
 
@@ -868,6 +878,12 @@ TEST_P(Test_ONNX_layers, PoolConv3D)
     if (backend == DNN_BACKEND_VKCOM)
         applyTestTag(CV_TEST_TAG_DNN_SKIP_VULKAN);
 
+    if (backend == DNN_BACKEND_CUDA && target == DNN_TARGET_CUDA_FP16)
+    {
+        // CUDA_FP16: cuDNN did not return a suitable algorithm for convolution.
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_CUDA_FP16);
+    }
+
     testONNXModels("pool_conv_3d");
 }
 
@@ -1073,10 +1089,9 @@ TEST_P(Test_ONNX_layers, Div)
     Mat out = net.forward();
 
     normAssert(ref, out, "", default_l1,  default_lInf);
-    expectNoFallbacksFromIE(net);
-    expectNoFallbacksFromCUDA(net);
 
-    testONNXModels("div_test_1x1",npy, 0, 0, false, true, 2);
+    // NaryEltwise layer suuports only CPU for now
+    testONNXModels("div_test_1x1", npy, 0, 0, false, false, 2);
 }
 
 TEST_P(Test_ONNX_layers, DynamicReshape)
@@ -1122,8 +1137,17 @@ TEST_P(Test_ONNX_layers, Split)
     testONNXModels("split_2");
     testONNXModels("split_3");
     testONNXModels("split_4");
-    testONNXModels("split_sizes");
     testONNXModels("split_neg_axis");
+}
+
+// Mul inside with 0-d tensor, output should be A x 1, but is 1 x A. PR #22652
+TEST_P(Test_ONNX_layers, DISABLED_Split_sizes_0d)
+{
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE_NN_BUILDER_2019)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_NN_BUILDER);
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE_NGRAPH)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_NGRAPH);
+    testONNXModels("split_sizes");
 }
 
 TEST_P(Test_ONNX_layers, Slice)
@@ -2179,7 +2203,7 @@ TEST_P(Test_ONNX_nets, LResNet100E_IR)
     }
     else if (target == DNN_TARGET_CUDA_FP16)
     {
-        l1 = 0.008;
+        l1 = 0.009;
         lInf = 0.04;
     }
     testONNXModels("LResNet100E_IR", pb, l1, lInf);

